tests/integration/test_cos.py

```diff
@@ -0,0 +1,131 @@
+    if "microk8s" not in e.model.applications:
+        await e.model.deploy(
+            config.MK8S_CHARM,
+            application_name="microk8s",
+            config={"hostpath_storage": "true"},
+            channel=config.MK8S_CHARM_CHANNEL,
+            constraints=config.MK8S_CONSTRAINTS,
+        )
+        await e.model.wait_for_idle(["microk8s"])
+
+    if "grafana-agent" not in e.model.applications:
+        await e.model.deploy(
+            config.MK8S_GRAFANA_AGENT_CHARM,
+            channel=config.MK8S_GRAFANA_AGENT_CHANNEL,
+            application_name="grafana-agent",
+        )
+        await e.model.add_relation("microk8s", "grafana-agent")
+        await e.model.wait_for_idle(["microk8s", "grafana-agent"])
+
+    microk8s_unit: Unit = e.model.applications["microk8s"].units[0]
+    grafana_agent_app: Application = e.model.applications["grafana-agent"]
+
+    # TODO(neoaggelos): this is a terrible workaround but solves timeout issues in GitHub actions
+    await run_unit(
+        microk8s_unit,
+        "sudo sed -i 's,timeout: float = 5.0,timeout: float = 20.0,' /var/lib/juju/agents/unit-grafana-agent*/charm/lib/charms/operator_libs_linux/v1/snap.py",  # noqa
+    )
+
+    # bootstrap a juju cloud on the deployed microk8s
+    async with microk8s_kubernetes_cloud_and_model(e, "microk8s") as (k8s_model, k8s_model_name):
+        with e.model_context(k8s_model):
+            LOG.info("Deploy MetalLB")
+            await e.model.deploy(
+                config.MK8S_METALLB_CHARM,
+                application_name="metallb",
+                config={"iprange": "42.42.42.42-42.42.42.42"},
+                channel=config.MK8S_METALLB_CHANNEL,
+            )
+            await e.model.wait_for_idle(["metallb"])
+
+            LOG.info("Deploy cos-lite")
+            await e.model.deploy(
+                config.MK8S_COS_BUNDLE,
+                channel=config.MK8S_COS_CHANNEL,
+                trust=True,
+            )
+            await e.model.wait_for_idle(["prometheus", "grafana", "loki"], timeout=30 * 60)
+
+            LOG.info("Create offers for cos-lite endpoints")
+            await e.model.create_offer("prometheus:receive-remote-write", "prometheus")
+            await e.model.create_offer("loki:logging", "loki")
+            await e.model.create_offer("grafana:grafana-dashboard", "grafana")
+
+        try:
+            LOG.info("Consume cos-lite and relate with grafana-agent")
+            await e.model.consume(f"admin/{k8s_model_name}.prometheus", "prometheus")
+            await e.model.consume(f"admin/{k8s_model_name}.loki", "loki")
+            await e.model.consume(f"admin/{k8s_model_name}.grafana", "grafana")
+            await e.model.add_relation("grafana-agent", "prometheus")
+            await e.model.add_relation("grafana-agent", "loki")
+            await e.model.add_relation("grafana-agent", "grafana")
+
+            await e.model.wait_for_idle(["microk8s", "grafana-agent"], timeout=20 * 60)
```

Comment:
@sed-i here's where we get the timeout issues. pretty much, after relating to cos components, grafana-agent's hooks will check to see if the grafana-agent snap is installed, and that operation has a 5 second timeout (which is a bit low for the github runners).

Reply:
https://github.com/canonical/grafana-agent-k8s-operator/pull/247 might help with this